runtime/vm/ObjectMonitor.cpp

```diff
@@ -196,10 +196,12 @@ objectMonitorEnterBlocking(J9VMThread *currentThread)
 			/* In a Concurrent GC where monitor object can *move* in a middle of GC cycle,
 			 * we need a proper barrier to get an up-to-date location of the monitor object */
 			j9objectmonitor_t volatile *lwEA = VM_ObjectMonitor::inlineGetLockAddress(currentThread, J9MONITORTABLE_OBJECT_LOAD(currentThread, &((J9ThreadMonitor*)monitor)->userData));
-			j9objectmonitor_t lockInLoop = *lwEA;
+			// TODO: does macro break volatile?
+			j9objectmonitor_t lockInLoop = J9_LOAD_LOCKWORD(currentThread, lwEA);
 #if defined(J9VM_THR_LOCK_RESERVATION)
 			/* has the lock become reserved? */
-			if (OBJECT_HEADER_LOCK_RESERVED == (*lwEA & (OBJECT_HEADER_LOCK_RESERVED + OBJECT_HEADER_LOCK_INFLATED))) {
+			// TODO: Use lockInLoop instead of another read? */
```

Comment:
```suggestion
			/* TODO: Use lockInLoop instead of another read? */
```

Reply:
Also, why wouldn't you reuse the already fetched value here?

Reply:
Comment fixed. The previous code did not use the local, so the new code does not either (though I think it probably should - the TODO is left so this can be investigated later).